source/base/auto_derivative_function.cc

```diff
@@ -47,6 +47,19 @@ template <int dim>
 void
 AutoDerivativeFunction<dim>::set_formula (const DifferenceFormula form)
 {
+  // go through all known formulas, reject ones we don't know about
+  // and don't handle in the member functions of this class
+  switch (formula)
```

Comment:
should be `form`, not `formula`, in the switch. I had to look at that a few times.